### PR TITLE
Update the bit field of encoding function names

### PR DESCRIPTION
**Pull request by kunalspathak**
The arm manual parsing tool had a bug where I was not taking into account the extra space, resulting in function names off by 1. Renamed the function names.

### CODE REVIEW

**Comment by kunalspathak** [comment]
@dotnet/arm64-contrib @a74nh @SwapnilGaikwad

**Comment by SwapnilGaikwad** [comment]
LGTM!

File: src/coreclr/jit/emitarm64.cpp [hunk 1]
```diff
@@ -11589,28 +11589,28 @@ void emitter::emitIns_Call(EmitCallType          callType,
...
- *  Return an encoding for the specified 'V' register used in '21' thru '17' position.
+ *  Return an encoding for the specified 'V' register used in '20' thru '16' position.
  */
 
-/*static*/ emitter::code_t emitter::insEncodeReg_V_21_to_17(regNumber reg)
+/*static*/ emitter::code_t emitter::insEncodeReg_V_20_to_16(regNumber reg)
 {
     assert(isVectorRegister(reg));
     emitter::code_t ureg = (emitter::code_t)reg - (emitter::code_t)REG_V0;
     assert((ureg >= 0) && (ureg <= 32));
```
Comment (BruceForstall) [inline src/coreclr/jit/emitarm64.cpp:11599]: This is unrelated to your PR, but:
```suggestion
    assert((ureg >= 0) && (ureg <= 31));
```

File: src/coreclr/jit/emitarm64.cpp [hunk 2]
```diff
@@ -11589,28 +11589,28 @@ void emitter::emitIns_Call(EmitCallType          callType,
...
- *  Return an encoding for the specified 'R' register used in '21' thru '17' position.
+ *  Return an encoding for the specified 'R' register used in '20' thru '16' position.
  */
 
-/*static*/ emitter::code_t emitter::insEncodeReg_R_21_to_17(regNumber reg)
+/*static*/ emitter::code_t emitter::insEncodeReg_R_20_to_16(regNumber reg)
 {
     assert(isIntegerRegister(reg));
     emitter::code_t ureg = (emitter::code_t)reg;
     assert((ureg >= 0) && (ureg <= 32));
```
Comment (BruceForstall) [inline src/coreclr/jit/emitarm64.cpp:11612]:
```suggestion
    assert((ureg >= 0) && (ureg <= 31));
```

File: src/coreclr/jit/emitarm64.cpp [hunk 3]
```diff
@@ -11693,15 +11693,15 @@ void emitter::emitIns_Call(EmitCallType          callType,
...
- *  Return an encoding for the specified 'R' register used in '18' thru '17' position.
+ *  Return an encoding for the specified 'R' register used in '17' thru '16' position.
  */
 
-/*static*/ emitter::code_t emitter::insEncodeReg_R_18_to_17(regNumber reg)
+/*static*/ emitter::code_t emitter::insEncodeReg_R_17_to_16(regNumber reg)
 {
     assert(isIntegerRegister(reg));
     emitter::code_t ureg = (emitter::code_t)reg;
     assert((ureg >= 0) && (ureg <= 32));
```
Comment (BruceForstall) [inline src/coreclr/jit/emitarm64.cpp:11703]:
```suggestion
    assert((ureg >= 0) && (ureg <= 3));
```
?

File: src/coreclr/jit/emitarm64.cpp [hunk 4]
```diff
@@ -11755,28 +11755,28 @@ void emitter::emitIns_Call(EmitCallType          callType,
...
- *  Return an encoding for the specified 'V' register used in '19' thru '17' position.
+ *  Return an encoding for the specified 'V' register used in '18' thru '16' position.
  */
 
-/*static*/ emitter::code_t emitter::insEncodeReg_V_19_to_17(regNumber reg)
+/*static*/ emitter::code_t emitter::insEncodeReg_V_18_to_16(regNumber reg)
 {
     assert(isVectorRegister(reg));
     emitter::code_t ureg = (emitter::code_t)reg - (emitter::code_t)REG_V0;
     assert((ureg >= 0) && (ureg <= 32));
```
Comment (BruceForstall) [inline src/coreclr/jit/emitarm64.cpp:11765]:
```suggestion
    assert((ureg >= 0) && (ureg <= 7));
```
?

File: src/coreclr/jit/emitarm64.cpp [hunk 5]
```diff
@@ -11755,28 +11755,28 @@ void emitter::emitIns_Call(EmitCallType          callType,
...
- *  Return an encoding for the specified 'V' register used in '20' thru '17' position.
+ *  Return an encoding for the specified 'V' register used in '19' thru '16' position.
  */
 
-/*static*/ emitter::code_t emitter::insEncodeReg_V_20_to_17(regNumber reg)
+/*static*/ emitter::code_t emitter::insEncodeReg_V_19_to_16(regNumber reg)
 {
     assert(isVectorRegister(reg));
     emitter::code_t ureg = (emitter::code_t)reg - (emitter::code_t)REG_V0;
     assert((ureg >= 0) && (ureg <= 32));
```
Comment (BruceForstall) [inline src/coreclr/jit/emitarm64.cpp:11778]:
```suggestion
    assert((ureg >= 0) && (ureg <= 15));
```

**Review by BruceForstall — APPROVED** [review]

File: src/coreclr/jit/emitarm64.cpp (hunk 3 above)
Comment (kunalspathak) [reply src/coreclr/jit/emitarm64.cpp:11703]: `psel` uses this encoding and I am not sure what register represents by 2 bits. The [manual says](https://docsmirror.github.io/A64/2023-06/psel_p_ppi.html) "Is the 32-bit name of the vector select register W12-W15, encoded in the "Rv" field.". @a74nh do you know?

Comment (a74nh) [reply src/coreclr/jit/emitarm64.cpp:11703]: `PSEL <Pd>, <Pn>, <Pm>.<T>[<Wv>, <imm>]`

It's the `Wv` in the above. That's quite an odd instruction encoding - it's trying to fit in 4 registers, a 5bit immediate and the element size.

It's not explicitly mentioned, but `w12` would be `00`, `w13` would be `01`, etc  

So, the assert should be `(ureg >= 12) && (ureg <= 15)`.

Comment (kunalspathak) [reply src/coreclr/jit/emitarm64.cpp]: hoping to get it address in https://github.com/dotnet/runtime/pull/95996 to skip a CI run.

**Pull request merged**